vector/src/main/java/im/vector/app/features/voicebroadcast/listening/VoiceBroadcastPlayerImpl.kt

```diff
@@ -213,18 +214,17 @@ class VoiceBroadcastPlayerImpl @Inject constructor(
                     if (sequencePosition > 0) {
                         mp.seekTo(sequencePosition)
                     }
-                    playingState = State.PLAYING
+                    playingState = State.Playing
                     prepareNextMediaPlayer()
                 }
-            } catch (failure: Throwable) {
-                Timber.e(failure, "## Voice Broadcast | Unable to start playback: $failure")
-                throw VoiceFailure.UnableToPlay(failure)
+            } catch (failure: VoiceBroadcastFailure.ListeningError.DownloadError) {
+                playingState = State.Error(failure)
             }
         }
     }
 
     private fun pausePlayback() {
-        playingState = State.PAUSED // This will trigger a playing state update and save the current position
+        playingState = State.Paused // This will trigger a playing state update and save the current position
         if (currentMediaPlayer != null) {
             currentMediaPlayer?.pause()
```

Comment:
I see there is no `tryOrNull` on `currentMediaPlayer?.pause()` or `currentMediaPlayer?.start()`. Was it done on purpose?

Reply:
Each action on the media player can trigger an exception if the player is in a bad state. The exceptions can be triggered because of a bad implementation or an internal player error (eg. a corrupted file). For the moment, I never observed calls to play/pause within a bad player state, the exceptions that I observed have been triggered when `preparing` the player, they're already caught and the play/pause actions should not be available. If the player enters in a bad state for another reason, it should also be caught by the error listener.
Do you think it would be safer to try/catch these actions or do you think I miss something when I intercept the error?

Reply:
I think it is okay. I just wanted to understand why it is only done in some methods. In the doc it says `pause()` may throw exception only `if the internal player engine has not been initialized.` ans `start()` may throw exception `if it is called in an invalid state`. And I think right now, it is not possible to be in these situations.